### PR TITLE
MSSQL DBMS supports setting connection policy

### PR DESCRIPTION
**Pull request by zhongyi-zhang**
Solve https://github.com/Azure/open-service-broker-azure/issues/646 and do some small refactors by the way.

### CODE REVIEW

File: docs/modules/mssql.md [hunk 1]
```diff
@@ -49,6 +49,7 @@ Provisions a new SQL Server and a new database upon that server. The new dbms an
 | `firewallRules[n].name` | `string` | Specifies the name of the generated firewall rule |Y | |
 | `firewallRules[n].startIPAddress` | `string` | Specifies the start of the IP range allowed by this firewall rule | Y | |
 | `firewallRules[n].endIPAddress` | `string` | Specifies the end of the IP range allowed by this firewall rule | Y | |
+| `connectionPolicy` | `string` | Refer [here](https://docs.microsoft.com/en-us/azure/sql-database/sql-database-connectivity-architecture#connection-policy) to specify a connection policy. Valid values are "Redirect", "Proxy", "Default" | N | |
```
Comment (norshtein) [inline docs/modules/mssql.md]: For provisioning step, a default value should be specified. The same below.

Comment (zhongyi-zhang) [reply docs/modules/mssql.md]: It is an optional parameter. The provision step can be skip if the field is empty like an update step. Does it make sense?

File: pkg/services/mssql/plan_details.go [hunk 2]
```diff
@@ -281,71 +281,41 @@ func getDBMSCommonUpdateParamSchema() service.InputParametersSchema {
 					},
 				},
 			},
+			"connectionPolicy": &service.StringPropertySchema{
+				Title:       "Connection Policy",
+				Description: "The type of server connection policy.",
+				AllowedValues: []string{
+					"", // Allow empty. It doesn't call the setting method.
+					ConnectionPolicyDefault,
```
[the remark this answers is not in the document]
Comment (zhongyi-zhang) [reply pkg/services/mssql/plan_details.go]: Yeah, you're right.

File: docs/modules/mssql.md (hunk 1 above)
Comment (norshtein) [reply docs/modules/mssql.md]: It's developer's knowledge, however, if I'm a new user, I'll be confused about what will happen if I don't provide this non-required value, a brief clarification will be better.

Comment (zhongyi-zhang) [reply docs/modules/mssql.md]: Good point thanks. Updated now.

**Pull request merged**